src/main/scala/com/rbmhtechnology/calliope/serializer/kafka/PayloadFormatSerializer.scala

```diff
@@ -48,22 +51,46 @@ class PayloadFormatSerializer[A <: AnyRef] private(serializer: PayloadSerializer
 }
 
 object PayloadFormatDeserializer {
-  def apply[A](payloadMapper: AnyRef => A)(implicit system: ActorSystem): PayloadFormatDeserializer[A] =
-    new PayloadFormatDeserializer[A](DelegatingStringManifestPayloadSerializer(system), payloadMapper)
 
   def apply()(implicit system: ActorSystem): PayloadFormatDeserializer[AnyRef] =
     apply[AnyRef](identity)
 
-  def create[A](payloadMapper: JFunction[AnyRef, A], system: ActorSystem): PayloadFormatDeserializer[A] =
-    apply(payloadMapper.apply)(system)
+  def apply[A](payloadMapper: AnyRef => A)(implicit system: ActorSystem): PayloadFormatDeserializer[A] =
+    instance(payloadMapper, err => throw err)
+
+  def option[A](payloadMapper: PartialFunction[AnyRef, A])(implicit system: ActorSystem): PayloadFormatDeserializer[Option[A]] =
+    apply(payloadMapper.lift)
+
+  def filterError[A](payloadMapper: PartialFunction[AnyRef, A], onError: Throwable => Unit)(implicit system: ActorSystem): PayloadFormatDeserializer[Option[A]] =
```

Comment:
I think one `attempt` method should suffice as you can use `collect` to unpack and filter the elements within the stream of `Try[Option[A]]`. e.g. 
```
Source.empty[Try[Option[String]]]
  .collect {
    case Success(Some(payload)) => payload
  }
```
The same is be possible with the Java-API by using filter & map.

So we would have the following factory methods:
Scala:
` def attempt[A](payloadMapper: AnyRef => A)(implicit system: ActorSystem): PayloadFormatDeserializer[Try[A]]`

Java:
`def createAttempt[A](payloadMapper: JFunction[AnyRef, A])(implicit system: ActorSystem): PayloadFormatDeserializer[vavr.Try[A]]`